cli/test.sh

```diff
@@ -15,39 +15,43 @@
 
 set -eu
 
-binary="$(realpath ${1?})"
+binary="$(dirname "$1")/$(basename "$1")"
 iterations="${2:-0}"
 
-if [[ ! -x "$binary" ]]; then
+if ! test -x "$binary"; then
   echo "ERROR: $binary is not an executable"
   exit 1
 fi
 
-if (( iterations < 0 || iterations > 255 )); then
+if [ "$iterations" -lt 0 ] || [ "$iterations" -gt 255 ]; then
   echo "ERROR: the number of iterations must be within 0..255 (was: $iterations)"
   exit 2
 fi
 
-errors=0
-t=$(mktemp -d glometest.XXX)
-trap "rm -rf -- '${t?}'" EXIT
+t=$(mktemp -d)
+cleanup() {
+  rm -rf -- "${t?}"
+}
+trap cleanup EXIT
 
 for side in 0 1; do
         "$binary" genkey | tee "${t}/${side}" | "$binary" pubkey >"${t}/${side}.pub"
 done
 
+errors=0
 for counter in $(seq 0 "$iterations"); do
-  msg="$RANDOM"
+  msg="$(head -c 2 /dev/urandom | od -t u2 -A n)"
```

Comment:
I think the od is not obvious and could use a comment.

Reply:
I replaced that message generator with a static string because of (a) determinism and (b) bugfix (described in the commit message).